### PR TITLE
Change type of width & height from int to string in order to handle dimensions in other units than 'px' (i.e. '%').

### PR DESCRIPTION
**Pull request by mirtoto**
- Handle dimensions (width & height) in other units than 'px' (ie. '%').
- For backward compatibility, values without units are treated as 'px'.
- The default width values are now set to "100%".
- Add CSS class name to embedded objects.

### CODE REVIEW

**Comment by marekdedic** [comment]
Hi,
thanks for the PR, I will look at it and let you know 🙂

File: plugin/public/block.php [hunk 1]
```diff
@@ -313,11 +334,11 @@ function build_result( $content, $type, $arg ) {
 	if ( 'list' === $type ) {
 		//build list table
 		if ( ! empty( $arg ) ) {
-			$result = '<table style="width:' . $arg['width'] . 'px"><tbody>';
-		} else {
-			$result = '<table><tbody>';
+			$style = ' style="width:' . parse_dimension( $arg['width'] ) . ';"';
 		}
 
+		$result = '<table class="sgdd-embedded-folder-list"' . $style . '><tbody>';
```
Comment (marekdedic) [inline plugin/public/block.php:340]: Hi, I'm getting

```php
Notice: Undefined variable: style in /home/user/www/wp-content/plugins/skaut-google-drive-documents/public/block.php on line 340
```

Could you please look at it?

File: plugin/admin/options/class-options.php [hunk 2]
```diff
@@ -106,12 +106,12 @@ public static function init() {
 
 		self::$root_path = new \Sgdd\Admin\Options\OptionTypes\PathField( 'root_path', '', 'basic', 'path_selection', [] );
 
-		self::$embed_width  = new \Sgdd\Admin\Options\OptionTypes\IntegerField( 'embed_width', __( 'Width', 'skaut-google-drive-documents' ), 'advanced', 'file', '800' );
-		self::$embed_height = new \Sgdd\Admin\Options\OptionTypes\IntegerField( 'embed_height', __( 'Height', 'skaut-google-drive-documents' ), 'advanced', 'file', '600' );
+		self::$embed_width  = new \Sgdd\Admin\Options\OptionTypes\StringField( 'embed_width', __( 'Width', 'skaut-google-drive-documents' ), 'advanced', 'file', '100%' );
+		self::$embed_height = new \Sgdd\Admin\Options\OptionTypes\StringField( 'embed_height', __( 'Height', 'skaut-google-drive-documents' ), 'advanced', 'file', '600' );
```
Comment (marekdedic) [inline plugin/admin/options/class-options.php]: Could we have the default `600px` please in order for it to be better understandable?

File: plugin/public/block.php [hunk 3]
```diff
@@ -66,6 +67,25 @@ function add_block() {
...
+ *  - '100px' -> '100px',
+ *  - '100%' -> '100%'.
+ * 
+ * @param $in Input string
+ * @return string If $in contains only digits append to it 'px' otherwise return unmodified $in
+ */
+function parse_dimension( $in ) {
+	$out = $in;
+
+	if ( ctype_digit( $in ) ) {
```
Comment (marekdedic) [inline plugin/public/block.php:82]: Wouldn't `is_numeric` be stricter?

Comment (marekdedic) [reply plugin/public/block.php:82]: Also, can't we do this before saving the setting to the DB as opposed to her on output? So that if you just put in `600` it would convert to `600px` and show that when you next open the settings?

File: plugin/public/block.php [hunk 4]
```diff
@@ -148,14 +168,14 @@ function display( $attr ) {
...
+			$size .= 'width:' . parse_dimension( $attr['embedWidth'] ) . '; ';
 		}
 
 		if ( isset( $attr['embedHeight'] ) ) {
-			$size .= 'height:' . $attr['embedHeight'] . 'px; ';
+			$size .= 'height:' . parse_dimension( $attr['embedHeight'] ) . '; ';
 		}
 
-		$result = '<iframe src="https://drive.google.com/file/d/' . $id . '/preview" style="' . $size . 'border:0;"></iframe>';
+		$result = '<iframe class="sgdd-embedded-file" src="https://drive.google.com/file/d/' . $id . '/preview" style="' . $size . 'border:0;"></iframe>';
```
Comment (marekdedic) [inline plugin/public/block.php:178]: Is this change related to the PR? I'm a bit confused...

**Review by marekdedic — COMMENTED** [review]
Hi,
thanks for the PR. LGTM except for some minor comments - could you please look at them? Going forward, I would like to avoid having users know what CSS dimensions they can use, but seeing the plugin as it is now, this is fine and would probably need to be reworked altogether to make that make sense...

File: plugin/public/block.php (hunk 4 above)
Comment (marekdedic) [reply plugin/public/block.php:178]: Ah, that's the fourth point, I see...

**Review by marekdedic — COMMENTED** [review]
~Deleted~

File: plugin/public/block.php (hunk 4 above)
Comment (marekdedic) [reply plugin/public/block.php:178]: Umm, then the question becomes "Why?" The class isn't used anywhere else...

File: plugin/admin/options/class-options.php (hunk 2 above)
Comment (mirtoto) [reply plugin/admin/options/class-options.php]: 100% for nearly any situation will be proper value - embedded document just fill parent container. In general width expressed in pixels is against responsive design (usually it is a big problem for mobile devices which have much smaller screens than regular computers). In present time webpage should render on any screen size properly.

File: plugin/public/block.php (hunk 4 above)
Comment (mirtoto) [reply plugin/public/block.php:178]: Yes - it is not using by plugin :) It is possibility for plugin user for customization - i.e. add some margin. Without class name there is no chance to customize look of frame via external CSS.

File: plugin/public/block.php (hunk 1 above)
Comment (mirtoto) [reply plugin/public/block.php:340]: What tool displays such message?

File: plugin/admin/options/class-options.php (hunk 2 above)
Comment (marekdedic) [reply plugin/admin/options/class-options.php]: Oh, I meant the height :)

File: plugin/public/block.php (hunk 4 above)
Comment (marekdedic) [reply plugin/public/block.php:178]: Ok, makes sense now.

File: plugin/public/block.php (hunk 1 above)
Comment (marekdedic) [reply plugin/public/block.php:340]: Just the browser/server with `display_errors` turned on...

I think the error is on lines 328 and 336 which together may result in `$style` never being initialized when `$arg` is empty...

File: plugin/public/block.php [hunk 5]
```diff
@@ -305,6 +325,7 @@ function fetch_folder_content( $folder_id ) {
  */
 function build_result( $content, $type, $arg ) {
 	$result;
+	$style;
```
Comment (marekdedic) [inline plugin/public/block.php]:
```suggestion
	$style = "";
```

File: plugin/public/block.php (hunk 3 above)
Comment (mirtoto) [reply plugin/public/block.php:82]: No - check PHP documentation & examples:
- ctype_digit() return true only when all characters are digits,
- is_numeric() return true when characters are numeric in general, i.e. also for: '1337e0', '1.9', etc.

I wrote this function for backward compatibility (for existing plugin users which may have defined some values in DB already). In general it may be good idea to convert this width value before saving it to DB but it is not resolving problem of backward compatibility.

File: plugin/public/block.php (hunk 1 above)
Comment (mirtoto) [reply plugin/public/block.php:340]: Strange - I have `error_reporting=E_ALL` & `display_errors=On` and I don't see such notice.

File: plugin/public/block.php (hunk 3 above)
Comment (marekdedic) [reply plugin/public/block.php:82]: Ok, you're right about both. :)

File: plugin/public/block.php (hunk 1 above)
Comment (marekdedic) [reply plugin/public/block.php:340]: Is your `$arg` empty?

File: plugin/admin/options/class-options.php (hunk 2 above)
Comment (mirtoto) [reply plugin/admin/options/class-options.php]: Ok - I changed default height from 600 to 600px.

File: plugin/public/block.php (hunk 1 above)
Comment (mirtoto) [reply plugin/public/block.php:340]: Yes - I tested case with empty `$arg`. Anyway `$style = ''` should resolve this problem.

**Review by marekdedic — APPROVED** [review]
